x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/add_exception.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiContextMenuItem, EuiText } from '@elastic/eui';
+import React from 'react';
+import * as i18n from '../translations';
+
+interface AddExceptionProps {
+  disabled?: boolean;
+  eventKind?: string | null | undefined;
+  onClick: () => void;
+  ruleId: string | null | undefined;
+}
+
+const AddExceptionComponent: React.FC<AddExceptionProps> = ({
+  disabled,
+  eventKind,
+  onClick,
+  ruleId,
+}) => {
+  return eventKind === 'event' && ruleId ? (
```

Comment:
I see the `eventKind` and `ruleId` props are only needed to check if the component has to render or not, and are not used anywhere else. This suggests to me that this condition should be done where this component is rendered, so we don't need these two props here, and this component do not have to know about this logic.